streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -230,6 +230,8 @@ public void onPartitionsRevoked(final Collection<TopicPartition> assignment) {
                 // suspend active tasks
                 suspendTasksAndState();
             } catch (final Throwable t) {
+                // TODO: when this happens, we should still make sure the standby / active tasks list gets updated before returning
```

Comment:
Any exception thrown in the `onXX` callback will be swallowed as a log entry by `AbstractCoordinator`, that is why we need to remember the exception and re-throw it later in the next poll call.